src/badguy/flame.cpp

```diff
@@ -128,15 +125,20 @@ Flame::active_update(float dt_sec)
   if (m_type == ICE)
     m_sprite->set_angle(math::degrees(angle) * 3.0f);
 
-  if (m_sprite->get_action() == "fade" && m_sprite->animation_done())
+  if ((m_sprite->get_action() == "fade" || m_sprite->get_action() == "greyscale_fade") && m_sprite->animation_done())
```

Comment:
Instead of checking for the name of the action, you could have a state class member boolean that says if the flame is fading or not.